From 6c44fde37ad1660c55a7b360a3b034d58f348d8a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 22 Sep 2014 12:05:26 -0400
Subject: [tools] perf/symbols: Don't demangle parameters and such by default

Message-id: <1411387527-31226-2-git-send-email-jolsa@redhat.com>
Patchwork-id: 96425
O-Subject: [PATCH RHEL7.1 BZ1140522 1/2] perf symbols: Don't demangle parameters and such by default
Bugzilla: 1140522
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Petr Holasek <pholasek@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Namhyung Kim <namhyung@kernel.org>

Bugzilla: 1140522
https://bugzilla.redhat.com/show_bug.cgi?id=1140522

upstream
========
commit e71e79457b79a52827039d9d7f253321bfd342bd
Author: Namhyung Kim <namhyung@kernel.org>
Date:   Thu Jul 31 14:47:42 2014 +0900

description
===========
Some C++ symbols have very long name and they make column length longer.
Most of them are about parameters including templates and we can ignore
such info most of time IMHO.

This patch passes DMGL_NO_OPTS by default when calling bfd_demangle().
One can still see full symbols with -v/--verbose option.

before:
  JS_CallFunctionValue(JSContext*, JSObject*, JS::Value, unsigned int, JS::Value*, JS::Value*)

after:
  JS_CallFunctionValue

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/symbol-elf.c b/tools/perf/util/symbol-elf.c
index 6864661..b0c8e98 100644
--- a/tools/perf/util/symbol-elf.c
+++ b/tools/perf/util/symbol-elf.c
@@ -935,8 +935,11 @@ new_symbol:
    * to it...
    */
   if (symbol_conf.demangle) {
-   demangled = bfd_demangle(NULL, elf_name,
-       DMGL_PARAMS | DMGL_ANSI);
+   int demangle_flags = DMGL_NO_OPTS;
+   if (verbose)
+    demangle_flags = DMGL_PARAMS | DMGL_ANSI;
+
+   demangled = bfd_demangle(NULL, elf_name, demangle_flags);
    if (demangled != NULL)
     elf_name = demangled;
   }
diff --git a/tools/perf/util/symbol.h b/tools/perf/util/symbol.h
index 615c752..e79fcf0 100644
--- a/tools/perf/util/symbol.h
+++ b/tools/perf/util/symbol.h
@@ -59,6 +59,7 @@ extern Elf_Scn *elf_section_by_name(Elf *elf, GElf_Ehdr *ep,
 #endif
 
 #ifndef DMGL_PARAMS
+#define DMGL_NO_OPTS     0              /* For readability... */
 #define DMGL_PARAMS      (1 << 0)       /* Include function args */
 #define DMGL_ANSI        (1 << 1)       /* Include const, volatile, etc */
 #endif
-- 
1.7.1